### PR TITLE
8276806: Use Objects.checkFromIndexSize where possible in java.base

### PR DESCRIPTION
**Pull request by stsypanov**
This is a follow-up for https://github.com/openjdk/jdk/pull/4507, looks like there are some cases that were not covered.

Also this is related to https://github.com/openjdk/jdk/pull/3615

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8276806](https://bugs.openjdk.java.net/browse/JDK-8276806): Use Objects.checkFromIndexSize where possible in java.base

### Reviewers
 * [Roger Riggs](https://openjdk.java.net/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6297/head:pull/6297` \
`$ git checkout pull/6297`

Update a local copy of the PR: \
`$ git checkout pull/6297` \
`$ git pull https://git.openjdk.java.net/jdk pull/6297/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6297`

View PR using the GUI difftool: \
`$ git pr show -t 6297`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6297.diff">https://git.openjdk.java.net/jdk/pull/6297.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/io/ObjectInputStream.java [hunk 1]
```diff
@@ -1199,9 +1196,7 @@ public void readFully(byte[] buf) throws IOException {
      */
     public void readFully(byte[] buf, int off, int len) throws IOException {
         int endoff = off + len;
-        if (off < 0 || len < 0 || endoff > buf.length || endoff < 0) {
-            throw new IndexOutOfBoundsException();
-        }
+        Objects.checkFromToIndex(off, endoff, buf.length);
```
Comment (dfuch) [inline src/java.base/share/classes/java/io/ObjectInputStream.java:1199]: Why not using `Objects.checkFromIndexSize(off, len, buf.length);` just like in the previous change at line 1029 above?

Comment (stsypanov) [reply src/java.base/share/classes/java/io/ObjectInputStream.java]: Good point, done!

**Review by RogerRiggs — APPROVED** [review]

**Review by LanceAndersen — APPROVED** [review]
This Looks OK to me

**Pull request closed without merging**